geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/LiveKafkaConsumerFeatureSource.scala

```diff
@@ -114,19 +123,79 @@ class LiveKafkaConsumerFeatureSource(entry: ContentEntry,
     }, 0, 1, TimeUnit.SECONDS)
   }
 
-  override def run(): Unit = while (running.get) {
-    queue.take() match {
-      case update: CreateOrUpdate => featureCache.createOrUpdateFeature(update)
-      case del: Delete            => featureCache.removeFeature(del)
-      case clr: Clear             => featureCache.clear()
-      case m                      => throw new IllegalArgumentException(s"Unknown message: $m")
+  override def run(): Unit =
+    while (running.get) {
+      queue.take() match {
+        case update: CreateOrUpdate =>
+          featureCache.createOrUpdateFeature(update)
+          fireEvent(new KafkaFeatureEvent(this,
+                                          Type.CHANGED,
+                                          KafkaFeatureEvent.buildBounds(update.feature),
+                                          update.feature))
+
+        case del: Delete            =>
+          featureCache.removeFeature(del)
+          fireEvent(new FeatureEvent(this,
+                                     Type.REMOVED,
+                                     KafkaFeatureEvent.buildBounds(featureCache.features(del.id).sf),
+                                     KafkaFeatureEvent.buildId(del.id)))
+
+        case clr: Clear             =>
+          featureCache.clear()
+          fireEvent(new FeatureEvent(this,
+                                     Type.REMOVED,
+                                     KafkaConsumerFeatureSource.wholeWorldBounds,
+                                     Filter.INCLUDE))
+        case m                      => throw new IllegalArgumentException(s"Unknown message: $m")
+      }
     }
-  }
 
   // optimized for filter.include
   override def getCountInternal(query: Query): Int = featureCache.size(query.getFilter)
 
   override def getReaderForFilter(f: Filter): FR = featureCache.getReaderForFilter(f)
+
+  override def canEvent: Boolean = true
+
+  // Lazily fires events.
+  def fireEvent(event: => FeatureEvent) = {
+    if (contentState.hasListener) {
+      contentState.fireFeatureEvent(event)
+    }
+  }
+}
+
+import org.locationtech.geomesa.kafka.KafkaFeatureEvent._
+
+class KafkaFeatureEvent(source: AnyRef,
+                        eventType: FeatureEvent.Type,
+                        bounds: ReferencedEnvelope,
+                        val feature: SimpleFeature)
+  extends FeatureEvent(source, eventType, bounds, buildId(feature.getID)) { }
+
+object KafkaFeatureEvent {
```

Comment:
Suggestion: add 3 factory methods: 
```
KafkaFeatureEvent.changed(src, feature): FeatureEvent
KafkaFeatureEvent.removed(src, feature): FeatureEvent
KafkaFeatureEvent.cleared(src): FeatureEvent
```
Or something like that.  That will make run() more readable.